Detections/SigninLogs/SuccessfulAccount-SigninAttemptsByIPviaDisabledAccounts

```diff
@@ -0,0 +1,48 @@
+name: Sign-ins from IPs that attempt sign-ins to disabled accounts
+description: |
+  'Identifies IPs with failed attempts to sign in to one or more disabled accounts signed in successfully to another account.
+  References: https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/reference-sign-ins-error-codes
+  50057 - User account is disabled. The account has been disabled by an administrator.' This analytic will additionally identify the successful signed in accounts as the mapped account entities for investigation in Sentinel.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - Persistence
+relevantTechniques:
+  - T1078
+  - T1098
+query: |
+ let lookBack = 1d;
+  SigninLogs 
+  | where TimeGenerated >= ago(lookBack)
+  | where ResultType == "50057" 
+  | where ResultDescription == "User account is disabled. The account has been disabled by an administrator." 
+  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), disabledAccountLoginAttempts = count(), 
+  disabledAccountsTargeted = dcount(UserPrincipalName), applicationsTargeted = dcount(AppDisplayName), disabledAccountSet = makeset(UserPrincipalName), 
+  applicationSet = makeset(AppDisplayName) by IPAddress
+  | order by disabledAccountLoginAttempts desc
+  | join kind= leftouter (
+      // Consider these IPs suspicious - and alert any related  successful sign-ins
+      SigninLogs
+      | where TimeGenerated >= ago(lookBack)
+      | where ResultType == 0
+      | summarize successfulAccountSigninCount = dcount(UserPrincipalName), successfulAccountSigninSet = makeset(UserPrincipalName, 15) by IPAddress
+      // Assume IPs associated with sign-ins from 100+ distinct user accounts are safe
+      | where successfulAccountSigninCount < 100
+  ) on IPAddress  
+  // IPs from which attempts to authenticate as disabled user accounts originated, and had a non-zero success rate for some other account
+  | where successfulAccountSigninCount != 0
+  | project StartTimeUtc, EndTimeUtc, IPAddress, disabledAccountLoginAttempts, disabledAccountsTargeted, disabledAccountSet, applicationSet, 
```

Comment:
Above the project, it might be good to notate if the success was before or after the failure.  Helps build context.  You could add a column that identifies this, such as | extend SuccessBeforeFailure = iff(SuccessStartTime < StartTime, true, false)

Reply:
You would need to add something like this in the project inside the join -
| summarize SuccessStartTime = min(TimeGenerated), ....

Reply:
@shainw While I see that I could add this "| extend successbeforefailure = iff(TimeGenerated < StartTime, true, false) " to show per successful login, I don't see how to format this in the project even using the summarize you recommend. Because the project shows a set of successful sign-ins I don't see how I could summarize this.

Reply:
Will take offline for potential later change, no requirement as of now.